File: Strivers/src/basicrecursion/factorialofnnumbers/FactorialOfNNumbers.java
package basicrecursion.factorialofnnumbers;

import java.util.ArrayList;

public class FactorialOfNNumbers {
    public static void main(String[] args) {
        ArrayList<Long> res = factorialNumbers(2);
        System.out.println(res);
    }

    static ArrayList<Long> res = new ArrayList();
    static long origNum = 0;

    static ArrayList<Long> factorialNumbers(long n) {
        // code here
        origNum = n;
        factorial(n);
        return res;
    }

    static Long factorial(long n) {
        if (n <= 0) return 1L;
        long fact = n * factorial(n - 1);
        if(fact <= origNum){
            res.add(fact);
        }
        return fact;
    }
}

File: Strivers/src/basicrecursion/factorialofnnumbers/FactorialOfNNumbers.kt
package basicrecursion.factorialofnnumbers

val res = ArrayList<Long>()
val origNum = 6L

fun main(args: Array<String>) {
    factorialOfNNumbers(origNum)
    println(res)
}

fun factorialOfNNumbers(n: Long): Long {
    if (n <= 0) return 1
    val fact = n * factorialOfNNumbers(n-1)
    if (fact <= origNum) {
        res.add(fact)
    }
    return fact
}